plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java

```diff
@@ -136,14 +136,51 @@ public TransportRegisterModelAction(
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<MLRegisterModelResponse> listener) {
-        User user = RestActionUtils.getUserContext(client);
         MLRegisterModelRequest registerModelRequest = MLRegisterModelRequest.fromActionRequest(request);
         MLRegisterModelInput registerModelInput = registerModelRequest.getRegisterModelInput();
+        if (registerModelInput.getModelGroupId() == null) {
+            mlModelGroupManager.validateUniqueModelGroupName(registerModelInput.getModelName(), ActionListener.wrap(modelGroups -> {
+                if (modelGroups != null
+                    && modelGroups.getHits().getTotalHits() != null
+                    && modelGroups.getHits().getTotalHits().value != 0) {
+                    String modelGroupIdOfTheNameProvided = modelGroups.getHits().getAt(0).getId();
+                    registerModelInput.setModelGroupId(modelGroupIdOfTheNameProvided);
+                    checkUserAccess(registerModelInput, listener, true);
+                } else {
+                    checkUserAccess(registerModelInput, listener, false);
+                }
+            }, e -> {
+                log.error("Failed to search model group index", e);
+                listener.onFailure(e);
+            }));
+        } else {
+            checkUserAccess(registerModelInput, listener, false);
+        }
+    }
+
+    private void checkUserAccess(
+        MLRegisterModelInput registerModelInput,
+        ActionListener<MLRegisterModelResponse> listener,
+        Boolean isModelNameAlreadyExisting
+    ) {
+        User user = RestActionUtils.getUserContext(client);
         modelAccessControlHelper
             .validateModelGroupAccess(user, registerModelInput.getModelGroupId(), client, ActionListener.wrap(access -> {
                 if (!access) {
-                    log.error("You don't have permissions to perform this operation on this model.");
-                    listener.onFailure(new IllegalArgumentException("You don't have permissions to perform this operation on this model."));
+                    if (isModelNameAlreadyExisting) {
```

Comment:
I don't know if we should be super verbose here. Because some body might be surprised, I gave a model name and why it's being used as model group. What do you think about this message idea:

```
The model name 'xyz' is unavailable because it is already in use by another model group (Group ID: 'group_id'), to which you don't have access. Since you did not provide a model group ID, the system defaults to using the model name to register a new model group. Please choose a different name to proceed.
```